examples/weather/unified_recipe/train.py

```diff
@@ -0,0 +1,358 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+import os
+import hydra
+import wandb
+import matplotlib.pyplot as plt
+import time
+import fsspec
+import zarr
+from omegaconf import DictConfig, OmegaConf, ListConfig
+from torch.nn.parallel import DistributedDataParallel
+from torch.cuda.amp import GradScaler
+from torch.optim.lr_scheduler import SequentialLR
+from torch import nn
+from tqdm import tqdm
+
+# Add eval to OmegaConf TODO: Remove when OmegaConf is updated
+OmegaConf.register_new_resolver("eval", eval)
+
+try:
+    from apex import optimizers
+except:
+    raise ImportError(
+        "training requires apex package for optimizer."
+        + "See https://github.com/nvidia/apex for install details."
+    )
+
+from modulus import Module
+from modulus.distributed import DistributedManager
+from modulus.utils import StaticCaptureTraining, StaticCaptureEvaluateNoGrad
+from modulus.launch.logging import (
+    LaunchLogger,
+    PythonLogger,
+    initialize_mlflow,
+    RankZeroLoggingWrapper,
+)
+from modulus.launch.utils import load_checkpoint, save_checkpoint
+from modulus.metrics.general.mse import batch_normalized_mse
+
+from seq_zarr_datapipe import SeqZarrDatapipe
+
+
+
+@hydra.main(version_base="1.2", config_path="conf", config_name="config")
+def main(cfg: DictConfig) -> None:
+
```

Comment:
Can we get more documentation on exactly what this function does, the logic/order of operations for users. 

For example: 
Why do we batch normalize?
What are stages?
Why do we do static capturing?

What options are configurable and what aren't?